pyproject.toml

```diff
@@ -215,6 +215,7 @@ disallow_incomplete_defs = true
 no_implicit_optional = true
 warn_redundant_casts = true
 warn_unused_ignores = true
+disable_error_code = ["type-arg"]
```

Comment:
Sorry I did not realise during our call, but this is a pretty big downside given our GETTSIM experience, actually. `type-arg` caught quite a few bugs there; would not be nice to disable it globally. Not sure whether `GenericFunction` would be better from @janosg's perspective? Maybe `DagsCallable` ? 

If we keep with the current approach, I'd actually favor `Callable[..., Any]` above this.

Reply:
Very good point. My view is that the user-facing API should then use `Callable[..., Any]`, so that users see a type hint constructed with something they know. Whether our internal functions then use `GenericFunction`, `AnyCallable`, `Fn`, or something else, I don't mind. However, in line with your last point, we could simply write `Callable[..., Any]` everywhere, and I think that would be fine. It is only a bit cluttered for the type hint of what was previously called `FunctionCollection`, and otherwise okay.

Reply:
Both sounds good to me!

Reply:
I'm also in favor of `Callable[..., Any]` but have no strong opinion.